pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java

```diff
@@ -350,6 +377,9 @@ private void setUpPinotController() {
       _realtimeSegmentsManager = null;
     }
 
+    // setup constraints
+    setupHelixClusterConstraints();
```

Comment:
We should not set this for Pinot controller but only for Helix controller

Reply:
But then no constraint is added during `PINOT_ONLY` mode?

Reply:
`PINOT_ONLY` mode controller does not serve as the Helix controller, so it should not add this constraint to the cluster.

Reply:
Got it. thanks